block/block_cache.go

```diff
@@ -0,0 +1,64 @@
+package block
+
+import (
+	"sync"
+
+	"github.com/dymensionxyz/dymint/types"
+)
+
+type Cache struct {
+	cache map[uint64]CachedBlock
+	sync.Mutex
+	lowestCachedBlockHeight  uint64
+	highestCachedBlockHeight uint64
+}
+
+func (m *Cache) AddBlockToCache(h uint64, b *types.Block, c *types.Commit) {
+	m.Lock()
+	defer m.Unlock()
+	m.cache[h] = CachedBlock{Block: b, Commit: c}
+
+	types.NumberOfAccumulatedBlocks.Set(float64(len(m.cache)))
+	if m.lowestCachedBlockHeight == 0 || h < m.lowestCachedBlockHeight {
+		m.lowestCachedBlockHeight = h
+		types.LowestPendingBlockHeight.Set(float64(h))
+	}
+	if h > m.highestCachedBlockHeight {
+		m.highestCachedBlockHeight = h
+		types.HighestReceivedBlockHeight.Set(float64(h))
+	}
+}
+
+func (m *Cache) DeleteBlockFromCache(h uint64) {
+	m.Lock()
+	defer m.Unlock()
+	delete(m.cache, h)
+
+	types.NumberOfAccumulatedBlocks.Set(float64(len(m.cache)))
+	if h == m.lowestCachedBlockHeight {
+		m.lowestCachedBlockHeight = h + 1
```

Comment:
you're assuming blocks in cache are by order, but you can have a cache with heights e.g [5,8,12]